src/ansiblelint/rules/schema.py

```diff
@@ -120,6 +165,9 @@ def matchyaml(self, file: Lintable) -> list[MatchError]:
                     tag=f"schema[{file.kind}]",
                 ),
             )
+
+        if not result:
+            result = super().matchyaml(file)
         return result
 
 
```

Comment:
That is missing tests, it needs at least two tests, one with playbook and one with a task file.

Reply:
I would also recommend making this easier to extend for other similar checks, where we want to verify that a specific field is one of list of generated values, maybe similar to:

```python
FIELD_CHECKS = {
"become_method": get_app().runtime.plugins.become.keys()
}
```


Also be sure that this is defined at module or class level and not inside the method being called.

The effective error message can be built from this dictionary, so we would not need to create new message for each field check that we add later.

Reply:
updated